### PR TITLE
DEV2-3148: handle chat proxy

### PR DESCRIPTION
**Pull request by yairco1990**

### CODE REVIEW

File: Common/src/main/java/com/tabnineCommon/chat/ChatBrowser.kt [hunk 1]
```diff
@@ -161,7 +156,28 @@ class ChatBrowser(messagesRouter: ChatMessagesRouter, private val project: Proje
...
+fun getProxySettings(): String? {
+    val httpConfigurable = HttpConfigurable.getInstance()
+
+    return if (httpConfigurable.USE_HTTP_PROXY) {
+        val auth = if (httpConfigurable.PROXY_AUTHENTICATION && httpConfigurable.proxyLogin?.isNotEmpty() == true) {
+            "${httpConfigurable.proxyLogin}:${httpConfigurable.plainProxyPassword}@"
+        } else {
+            ""
+        }
+        "${auth}${httpConfigurable.PROXY_HOST}:${httpConfigurable.PROXY_PORT}"
```
Comment (dimacodota) [inline Common/src/main/java/com/tabnineCommon/chat/ChatBrowser.kt:179]: I think this is missing the protocol(http/https)

**Review by dimacodota — COMMENTED** [review]
See my comment

File: Common/src/main/java/com/tabnineCommon/chat/ChatBrowser.kt (hunk 1 above)
Comment (yairco1990) [reply Common/src/main/java/com/tabnineCommon/chat/ChatBrowser.kt:179]: but there is no option in IJ settings to choose a protocol. by omitting the protocol, it should use http.

Comment (dimacodota) [reply Common/src/main/java/com/tabnineCommon/chat/ChatBrowser.kt:179]: Yes, we use http for the binary

Comment (dimacodota) [reply Common/src/main/java/com/tabnineCommon/chat/ChatBrowser.kt:179]: https://github.com/codota/tabnine-intellij/blob/1addb608344be27f228518a883420a22046f3e53/Common/src/main/java/com/tabnineCommon/binary/BinaryProcessGateway.kt#L56

Comment (yairco1990) [reply Common/src/main/java/com/tabnineCommon/chat/ChatBrowser.kt:179]: but what I did is part of the chromium protocol:
https://www.chromium.org/developers/design-documents/network-settings/#command-line-options-for-proxy-settings
we can set "http=...", but "http://..." doesn't seem to be the protocol in chromium. (http is the default so we don't have to specify it)

**Review by dimacodota — APPROVED** [review]

**Pull request merged**